python/dgl/contrib/multi_gpu_datastore.py

```diff
@@ -0,0 +1,156 @@
+##
+#   Copyright 2021 Contributors 
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+""" The MultiGPUDataStore class. """
+
+from .. import backend as F
+
+class MultiGPUDataStore:
+    """ Class for storing a large tensor split across GPU memory according to
+    nodes. When reading memory from other GPUs, the call must be from all
+    GPUs storing the data.
+
+    Example:
+    --------
+
+    It can be created via the shape and data type information from the tensor
+    to split across gpus `large_cpu_tensor` in the below code:
+
+    >>> split_data = MultiGPUTensor(large_cpu_tensor.shape,
+    ...                               large_cpu_tensor.dtype,
+    ...                               dev_id,
+    ...                               nccl_comm)
+    >>> split_data.set_global(large_cpu_tensor)
+
+    Then, once it is stored across GPU memory during training, features for
+    mini-batches can be fetched via the `get_global()` method. If we have the
+    tensor of mini-batch nodes `input_nodes` we can use the following code to
+    fetch the features for the mini-batch:
+    ...
+    >>> batch_features = split_data.all_gather_row(input_nodes)
+    """
+    def __init__(self, shape, dtype, device, comm, partition):
+        """ Create a new Tensor stored across multiple GPUs according to
+        `partition`. This function must be called by all processes.
+
+        Parameters
+        ----------
+        shape : tuple
+            The shape of the tensor. The tensor will be partitioned across its
+            first dimension. As a result, dimensionless tensors are not
+            allowed.
+        dtype : DType
+            The backend data type.
+        device : Device
+            The current backend device.
+        comm : nccl.Communicator
+            The NCCL communicator to use.
+        partition : NDArrayPartition
```

Comment:
The example code in the above does not pass in a partition object.

Reply:
Sorry, this is left over from the previous iteration. I've updated the docs to have a partition in this example in 
57b6a43